extension/observer/k8s/handler.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8s
+
+import (
+	"fmt"
+	"reflect"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+)
+
+// handler handles k8s cache informer callbacks.
+type handler struct {
+	// idNamespace should be some unique token to distinguish multiple handler instances.
+	idNamespace string
+	// watcher is the callback for discovered endpoints.
+	watcher observer.Notify
+}
+
+// OnAdd is called in response to a pod being added.
+func (h *handler) OnAdd(obj interface{}) {
+	pod := obj.(*v1.Pod)
```

Comment:
Is it guaranteed for `obj` to be `v1.Pod`? Perhaps be defensive and check?

Reply:
I would still be careful. I assume checking the interface has no or little cost. I get an eye pain from this unprotected conversion :-)
If for some reason you do not want to add the check I would document that obj is expected to be of particular interface and why the expectation.